astropy/units/core.py

```diff
@@ -2431,7 +2460,9 @@ def __copy__(self):
         """
         return CompositeUnit(self._scale, self._bases[:], self._powers[:])
 
-    def decompose(self, bases=set()):
+    def decompose(self, bases=None):
+        if bases is None:
```

Comment:
I think we should not just change things to do exactly the same thing but just use more cycles. Here, the next line clearly is checking whether `bases` is set, so can be `if bases is None and ...`. Similarly the `if len(bases) == 0` at the end could be worked into here.